source/service.go

```diff
@@ -138,21 +138,37 @@ func endpointsFromService(svc *v1.Service) []*endpoint.Endpoint {
 	var endpoints []*endpoint.Endpoint
 
 	// Get the desired hostname of the service from the annotation.
-	hostname, exists := svc.Annotations[hostnameAnnotationKey]
+	hostnameAnnotation, exists := svc.Annotations[hostnameAnnotationKey]
 	if !exists {
 		return nil
 	}
 
-	// Create a corresponding endpoint for each configured external entrypoint.
-	for _, lb := range svc.Status.LoadBalancer.Ingress {
-		if lb.IP != "" {
-			//TODO(ideahitme): consider retrieving record type from resource annotation instead of empty
-			endpoints = append(endpoints, endpoint.NewEndpoint(hostname, lb.IP, ""))
-		}
-		if lb.Hostname != "" {
-			endpoints = append(endpoints, endpoint.NewEndpoint(hostname, lb.Hostname, ""))
+	hostnames := parseHostnameAnnotation(hostnameAnnotation)
+
+	for _, hostname := range hostnames {
```

Comment:
maybe inline?

for _, hostname := range parseHostnameAnnotation(hostnameAnnotation) {